File: mall-order/src/main/java/com/hj/mall/order/entity/OrderItem.java
package com.hj.mall.order.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("oms_order_item")
public class OrderItem {

    @TableId(type = IdType.ASSIGN_ID)
    private Long id;

    private Long orderId;

    private String orderSn;

    private Long spuId;

    private Long skuId;

    private String spuName;

    private String skuName;

    private String skuImg;

    private BigDecimal skuPrice;

    private Integer skuQuantity;

    private BigDecimal skuTotalPrice;

    private BigDecimal promotionAmount;

    private BigDecimal couponAmount;

    private LocalDateTime createTime;

    // SPU图片
    private String spuPic;

    // SPU品牌
    private String spuBrand;

    // 分类ID
    private Long categoryId;

    // SKU属性值
    private String skuAttrsVals;

    // 积分抵扣金额
    private BigDecimal integrationAmount;

    // 实际支付金额
    private BigDecimal realAmount;

    // 赠送积分
    private Integer giftIntegration;

    // 赠送成长值
    private Integer giftGrowth;
}

File: mall-admin/src/api/order/orderItem.ts
import { get } from '../request'
import type { PageResult } from '../request'

export interface OrderItem {
  id?: number
  orderId?: number
  orderSn?: string
  skuId?: number
  skuName?: string
  skuImage?: string
  skuPrice: number
  skuQuantity: number
  totalPrice: number
}

export interface OrderItemParams {
  key?: string
  orderId?: number
  orderSn?: string
}

// 获取订单项列表
export const getOrderItemList = (params?: OrderItemParams) => {
  return get<PageResult<OrderItem>>('/admin/order/item/list', params)
}

// 根据订单 ID 获取订单项
export const getItemsByOrderId = (orderId: number) => {
  return get<OrderItem[]>(`/admin/order/item/order/${orderId}`)
}
